cpp/mg_utility/mage.hpp

```diff
@@ -0,0 +1,2757 @@
+  }
+  mgp::properties_iterator_destroy(properties_iterator);
+}
+
+inline Value Properties::operator[](const std::string_view key) const { return property_map_.at(key); }
+
+inline bool Properties::operator==(const Properties &other) const { return property_map_ == other.property_map_; }
+
+// Labels:
+
+inline std::string_view Labels::Iterator::operator*() const { return (*iterable_)[index_]; }
+
+inline std::string_view Labels::operator[](size_t index) const { return mgp::vertex_label_at(node_ptr_, index).name; }
+/* #endregion */
+
+/* #region Types */
+
+/* #region Containers (List, Map) */
+
+// List:
+
+inline List::List(const std::vector<Value> &values) : ptr_(mgp::list_make_empty(values.size(), memory)) {
+  for (const auto &value : values) {
+    AppendExtend(value);
+  }
+}
+
+inline List::List(std::vector<Value> &&values) : ptr_(mgp::list_make_empty(values.size(), memory)) {
+  for (auto &value : values) {
+    Append(std::move(value));
+  }
+}
+
+inline List::List(const std::initializer_list<Value> values) : ptr_(mgp::list_make_empty(values.size(), memory)) {
+  for (const auto &value : values) {
+    AppendExtend(value);
+  }
+}
+
+inline List::~List() {
+  if (ptr_ != nullptr) {
+    mgp::list_destroy(ptr_);
+  }
+}
+
+inline const Value List::operator[](size_t index) const { return Value(mgp::list_at(ptr_, index)); }
+
+inline Value List::Iterator::operator*() const { return (*iterable_)[index_]; }
+
+inline void List::Append(const Value &value) { mgp::list_append(ptr_, value.ptr_); }
+
+inline void List::Append(Value &&value) {
+  mgp::list_append(ptr_, value.ptr_);
+  value.ptr_ = nullptr;
+}
+
+inline void List::AppendExtend(const Value &value) { mgp::list_append_extend(ptr_, value.ptr_); }
+
+inline void List::AppendExtend(Value &&value) {
+  mgp::list_append_extend(ptr_, value.ptr_);
+  value.ptr_ = nullptr;
+}
+
+inline bool List::operator==(const List &other) const { return util::ListsEqual(ptr_, other.ptr_); }
+
+// MapItem:
+
+inline bool MapItem::operator==(MapItem &other) const { return key == other.key && value == other.value; }
+inline bool MapItem::operator!=(MapItem &other) const { return !(*this == other); }
+
+// Map:
+
+inline Map::Map(const std::map<std::string_view, Value> &items) : ptr_(mgp::map_make_empty(memory)) {
+  for (const auto &[key, value] : items) {
+    Insert(key, value);
+  }
+}
+
+inline Map::Map(std::map<std::string_view, Value> &&items) : ptr_(mgp::map_make_empty(memory)) {
+  for (auto &[key, value] : items) {
+    Insert(key, value);
+  }
+}
+
+inline Map::Map(const std::initializer_list<std::pair<std::string_view, Value>> items)
+    : ptr_(mgp::map_make_empty(memory)) {
+  for (const auto &[key, value] : items) {
+    Insert(key, value);
+  }
+}
+
+inline Map::~Map() {
+  if (ptr_ != nullptr) {
+    mgp::map_destroy(ptr_);
+  }
+}
+
+inline const Value Map::operator[](std::string_view key) const { return Value(mgp::map_at(ptr_, key.data())); }
```

Comment:
hmm this usually cannot be const because `operator[]` serves as both for setting values and viewing, if it is const it loses the purpose of setting value

Reply:
NB: This goes for both `List` and `Map`.

As setting map values is highly non-trivial in the existing C API, I’ve chosen to mark this operator as const and thus prohibit value-setting. This does make it equivalent to `at`, but I believe this API is better off with it because people are used to accessing maps with `[]`.